modules/set.js

```diff
@@ -0,0 +1,14 @@
+import isArray from './isArray.js';
+import isObject from './isObject.js';
+import _set from './_set.js';
+
+
+// set the value in given path
+export default function set (obj, path, value) {
+  if (!isObject(obj) || !isArray(path)) return obj;
+  if (path.length === 0) return obj;
```

Comment:
Sorry, I don't see other option better than this. I like this implementation. Please let me know if you have an idea how to improve it.

Reply:
Please refer to the approach I suggested in #2948. If you do postorder assignment instead of preorder assignment, you can check for `path.length === 0` only in the internal function and immediately return `value` if true. You can avoid lookahead beyond `path[0]` in that case by replacing `obj` instead of `obj[key]`.

Referring again to #2948, I also realize that your code is still vulnerable to prototype pollution. You need to fix that as well.